### PR TITLE
Headless/tahsin/order api/additional requirements

### PR DESCRIPTION
**Pull request by ghost**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [x] Link to a **GH Issue** or **Jira** task here for reference.
- [x] Bug fix
- [ ] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [ ] Separation of Concerns followed (Reviewer name/NaN)
- [ ] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/models/src/order/getAllOrderQuery.interface.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface GetAllOrderQuery {
+    shippingStatus?: string;
+    orderStatus?: string;
+    paymentStatus?: string;
+    skip?: number;
+    limit?: number;
+    startDate?: Date;
+    endDate?: Date;
+    startTime?: string;
+    endTime?: string
```
Comment (Anjarul-Islam) [inline packages/models/src/order/getAllOrderQuery.interface.ts]: front end will send starttime with startDate. actually full date-time can be taken as a single input value. no need to take start time and end time as different field.

File: yarn.lock [hunk 2]
```diff
@@ -10824,6 +10824,11 @@ moment-timezone@^0.5.34:
   resolved "https://registry.yarnpkg.com/moment/-/moment-2.29.3.tgz#edd47411c322413999f7a5940d526de183c031f3"
   integrity sha512-c6YRvhEo//6T2Jz/vVtYzqBzwvPT95JBQ+smCytzf7c50oMZRsR/a4w88aD34I+/QVSfnoAnSBFPJHItlOMJVw==
 
+moment@^2.29.4:
```
Comment (Anjarul-Islam) [inline yarn.lock:10827]: why and where are you using monent.js?

Comment (ghost) [reply yarn.lock:10827]: did not use it, but was going to use it for independent time filtering

Comment (ghost) [reply yarn.lock:10827]: will remove it

File: packages/headless/src/database/mongodb/order/index.ts [hunk 3]
```diff
@@ -5,6 +5,8 @@ import { OrderData } from 'src/modules/order/dto/order.response.dto';
 import { IOrderDatabase } from 'src/modules/order/repositories/order.db.interface';
 import { ProductModel } from '../product/product.model';
 import { OrderModel } from './order.model';
+import { GetAllOrderQueryDto } from 'src/modules/order/dto/allOrderList.dto';
+import moment from 'moment';
```
Comment (Anjarul-Islam) [inline packages/headless/src/database/mongodb/order/index.ts]: maybe unused moment import here

File: yarn.lock (hunk 2 above)
Comment (Anjarul-Islam) [reply yarn.lock:10827]: okay. so if you try a new package and dont you use it then you should remove the dependency. and also please check which files you are going to add for commit and what changes you’ve made there. i think if you check each file's change after adding to stage (git stage) and before commit, you may avoid these scenarios.

Comment (ghost) [reply yarn.lock:10827]: okay bhaia, sorry

**Comment by ghost** [comment]
Jira Link:  [https://pm23.atlassian.net/browse/BC-339](url)

File: packages/headless/src/modules/order/repositories/index.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
+import { GetAllOrderQueryDto } from 'src/modules/order/dto/allOrderList.dto';
 import { Injectable } from '@nestjs/common';
 import { randomInt } from 'crypto';
 
 import { ProductPhotoDto } from 'src/modules/product/dto/product.dto';
-import { ProductOrderDto, CreateOrderDto } from './../dto/order.create.dto';
+import { ProductOrderDto, CreateOrderDto } from 'src/modules/order/dto/order.create.dto';
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/order/repositories/index.ts]: why you’ve changed the path here? relative path would've been okay.

File: packages/headless/src/modules/order/services/admin.service.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import { AllOrderListDto, AllOrderResponseDto } from './../dto/allOrderList.dto';
-import { OrderResponseDto } from './../dto/order.response.dto';
+import { AllOrderListDto, AllOrderResponseDto, GetAllOrderQueryDto } from 'src/modules/order/dto/allOrderList.dto';
+import { OrderResponseDto } from 'src/modules/order/dto/order.response.dto';
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/order/services/admin.service.ts:2]: reference should be relative path

File: packages/headless/package.json [hunk 6]
```diff
@@ -33,19 +33,19 @@
     "@nestjs/swagger": "^5.2.1",
     "@nestjs/testing": "^8.4.7",
     "@nestjsi/class-validator": "^0.1.4",
-    "@types/sequelize": "^4.28.13",
```
Comment (Anjarul-Islam) [inline packages/headless/package.json:36]: types/sequelize is used twice. remove one. i'd suggest to copy the package.json from the base branch and hse that in your code. as you didn’t add/remove any dependency, no change should be in package json and yarn.lock file

**Pull request merged**